### ISSUE TITLE
Julia builds for `native` CPU, even with `XC_HOST = x86_64-w64-mingw32`

### ISSUE THREAD

**Issue by topolarity** [JuliaLang/julia#54808]
Following the instructions at https://github.com/JuliaLang/julia/blob/master/doc/src/devdocs/build/windows.md and then distributing to another machine (with a different architecture), I ran into:
```
ERROR: Unable to find compatible target in cached code image.
Target 0 (alderlake): Rejecting this target due to use of runtime-disabled features
```

It's not obvious to me which of `ARCH` / `MARCH` / `JULIA_CPU_TARGET` should need setting, but the cross-compiling from Unix instructions probably also behave this way.

**Comment by vchuravy** [comment]
You always need to set `JULIA_CPU_TARGET` if you want to be able to distribute the resulting binary?

**Comment by topolarity** [comment]
Yeah, this is mostly a documentation issue - the instructions specifically mention moving the result to another machine but never tell you to set `JULIA_CPU_TARGET`
